qualtran/bloqs/mod_arithmetic/mod_addition.py

```diff
@@ -279,12 +298,53 @@ class CModAddK(Bloq):
     def signature(self) -> 'Signature':
         return Signature([Register('ctrl', QBit()), Register('x', QUInt(self.bitsize))])
 
+    def build_composite_bloq(
+        self, bb: 'BloqBuilder', ctrl: 'Soquet', x: 'Soquet'
+    ) -> Dict[str, 'SoquetT']:
+        k = bb.add(IntState(bitsize=self.bitsize, val=self.k))
+        ctrl, k, x = bb.add(CModAdd(QUInt(self.bitsize), mod=self.mod), ctrl=ctrl, x=k, y=x)
+        bb.add(IntEffect(bitsize=self.bitsize, val=self.k), val=k)
+        return {'ctrl': ctrl, 'x': x}
+
+    def on_classical_vals(
+        self, ctrl: 'ClassicalValT', x: 'ClassicalValT'
+    ) -> Dict[str, 'ClassicalValT']:
+        if ctrl == 0:
+            return {'ctrl': 0, 'x': x}
+
+        assert ctrl == 1, 'Bad ctrl value.'
+        x = (x + self.k) % self.mod
+        return {'ctrl': ctrl, 'x': x}
+
     def build_call_graph(self, ssa: 'SympySymbolAllocator') -> 'BloqCountDictT':
-        k = ssa.new_symbol('k')
-        return {AddK(k=k, bitsize=self.bitsize).controlled(): 5}
+        return {CModAdd(QUInt(self.bitsize), mod=self.mod): 1}
 
-    def short_name(self) -> str:
-        return f'x += {self.k} % {self.mod}'
+    def wire_symbol(
+        self, reg: Optional['Register'], idx: Tuple[int, ...] = tuple()
+    ) -> 'WireSymbol':
+        if reg is None:
+            return Text(f"mod {self.mod}")
+        if reg.name == 'ctrl':
+            return Circle()
+        if reg.name == 'x':
+            return TextBox(f'x += {self.k} mod {self.mod}')
```

Comment:
I would suggest only putting the mod value in one place. These symbols are used in the "musical score" circuit diagram where width is at a premium. The `reg is None` case puts a little label _above_ the box. The register-specific ones put text _in_ the box. I'd ommit the mod from the x box